ssl/record/ssl3_record_tls13.c

```diff
@@ -143,7 +144,28 @@ int tls13_enc(SSL *s, SSL3_RECORD *recs, size_t n_recs, int sending)
     if (EVP_CipherInit_ex(ctx, NULL, NULL, NULL, iv, sending) <= 0
             || (!sending && EVP_CIPHER_CTX_ctrl(ctx, EVP_CTRL_AEAD_SET_TAG,
                                              taglen,
-                                             rec->data + rec->length) <= 0)
+                                             rec->data + rec->length) <= 0)) {
+        return -1;
+    }
+
+    /* Set up the AAD */
+    if (!WPACKET_init_static_len(&wpkt, recheader, sizeof(recheader), 0)
+            || !WPACKET_put_bytes_u8(&wpkt, rec->type)
+            || !WPACKET_put_bytes_u16(&wpkt, rec->rec_version)
+            || !WPACKET_put_bytes_u16(&wpkt, rec->length + taglen)
+            || !WPACKET_finish(&wpkt)) {
+        return -1;
```

Comment:
This seems to implicitly rely on knowledge of the structure of the record header/SSL3_RT_HEADER_LENGTH -- do we want to check that there's no space left in the WPACKET and/or that what we've written matches SSL3_RT_HEADER_LENGTH.  This seems relevant given that we add the AAD based on sizeof(recheader) and not the WPACKET's idea of things.

Reply:
Good point. I pushed a fixup commit to address this. Also realised we need to call `WPACKET_cleanup()` in the error case so I added that too.